tests/metrics/test_debias.py

```diff
@@ -0,0 +1,243 @@
+                Columns.Item: [1, 3, 1, 1, 2, 3, 4, 5, 1, 1, 2, 3, 1, 2, 1],
+                Columns.Rank: [9, 1, 3, 1, 3, 5, 7, 9, 1, 1, 2, 3, 2, 1, 1],
+            }
+        )
+        return reco_df
+
+    @pytest.fixture
+    def empty_interactions(self) -> pd.DataFrame:
+        return pd.DataFrame(columns=[Columns.User, Columns.Item], dtype=int)
+
+    @pytest.fixture
+    def catalog(self) -> tp.List[int]:
+        return list(range(10))
+
+    @pytest.fixture
+    def debias_config(self) -> DebiasConfig:
+        return DebiasConfig()
+
+    @pytest.fixture
+    def interactions_downsampling(self, interactions: pd.DataFrame, debias_config: DebiasConfig) -> pd.DataFrame:
+        return make_downsample(interactions, debias_config)
+
+    @pytest.fixture
+    def merged_downsampling(
+        self, interactions: pd.DataFrame, recommendations: pd.DataFrame, debias_config: DebiasConfig
+    ) -> pd.DataFrame:
+        merged = merge_reco(recommendations, interactions)
+        return make_downsample(merged, debias_config)
+
+    def test_make_downsample(
+        self, interactions: pd.DataFrame, recommendations: pd.DataFrame, debias_config: DebiasConfig
+    ) -> None:
+        merged = merge_reco(recommendations, interactions)
+
+        expected_result = pd.DataFrame(
+            {
+                Columns.User: [1, 1, 2, 3, 3, 3, 3, 3, 3, 5, 5, 5, 7],
+                Columns.Item: [2, 1, 1, 2, 3, 4, 5, 6, 1, 2, 3, 1, 1],
+            }
+        )
+        expected_result = pd.merge(
+            expected_result,
+            recommendations,
+            how="left",
+            on=Columns.UserItem,
+        )
+
+        interactions_downsampling = make_downsample(interactions, debias_config)
+        merged_downsampling = make_downsample(merged, debias_config)
+
+        pd.testing.assert_frame_equal(interactions_downsampling, expected_result[Columns.UserItem], check_like=True)
+        pd.testing.assert_frame_equal(merged_downsampling, expected_result, check_like=True)
+
+    def test_make_downsample_with_empty_data(
+        self, empty_interactions: pd.DataFrame, debias_config: DebiasConfig
+    ) -> None:
+        interactions_downsampling = make_downsample(empty_interactions, debias_config)
+
+        pd.testing.assert_frame_equal(interactions_downsampling, empty_interactions, check_like=True)
+
+    @pytest.mark.parametrize(
+        "metric",
+        (
+            Precision(k=1),
+            Precision(k=3),
+            Recall(k=1),
+            Recall(k=3),
+            F1Beta(k=1),
+            F1Beta(k=3),
+            Accuracy(k=1),
+            Accuracy(k=3),
+            MCC(k=1),
+            MCC(k=3),
+            HitRate(k=1),
+            HitRate(k=3),
+            MAP(k=1),
+            MAP(k=3),
+            NDCG(k=1),
+            NDCG(k=3),
+            MRR(k=1),
+            MRR(k=3),
+        ),
+    )
+    def test_debias_metric_calc(
```

Comment:
this and next tests should be in `test_scoring.py`

Reply:
Sorry, my bad.  
The point was that in `test_debias.py` should be only tests that test the code from `debias.py`. And if you want to test `Precision` metric with debias, the test should be in `test_classification.py` since `Precision` depends on debias, not vice versa

Reply:
I agree. I'll rewrite the tests

Reply:
**Done**